src/tool/abi/main.cpp

```diff
@@ -0,0 +1,179 @@
+#include "pch.h"
+
+#include "abi_writer.h"
+#include "common.h"
+#include "strings.h"
+
+using namespace std::chrono;
+using namespace std::experimental::filesystem;
+using namespace std::literals;
+using namespace xlang;
+using namespace xlang::meta::reader;
+using namespace xlang::text;
+using namespace xlang::cmd;
+
+auto output_directory(reader const& args)
+{
+    auto out{ absolute(args.value("output", "abi")) };
+    create_directories(out);
+    out += path::preferred_separator;
+    return out.string();
+}
+
+void print_usage()
+{
+    puts("Usage...");

```

Comment:
>puts("Usage..."); [](start = 4, length = 17)

don't forget

Reply:
I was wanting to follow existing format to be consistent, but the existing tools just do the same thing, so I copied :/

---
In reply to: [227974213](https://github.com/Microsoft/xlang/pull/51#discussion_r227974213) [](ancestors = 227974213)

Reply:
I think we'll probably just write a universal usage generator rather than maintaining various individual blurbs. This is a reasonable placeholder until we get that working.